### ISSUE TITLE
"User Rules for PAC" priority shall be higher than "GFW List"

### ISSUE THREAD

**Issue by yquan** [shadowsocks/ShadowsocksX-NG#317]
**System and Shadowsocksx-NG version:**

- macOS 10.12.4 (16E195)
- Shadowsocksx-NG Version 1.5.0 (3)

**Expected behavior:**

The "User List for PAC" shall be overriding the rules of "GFW List"

**Actual behavior:**

Currently, I found out that "User List for PAC" appended to "GFW List" in the generated PAC. This results no way to override any rules that are existed in GFW List.

**Comment by qiuyuzhou** [comment]
Very good to find that it should be the reason for most PAC problems.

**Comment by yquan** [comment]
Found this because it seems that https://fonts.gstatic.com is blocked and youtbue page doesn't display any characters at all while using safari by using the "Proxy Auto Configure Mode", while checking the pac file I found out following entry in GFW list:
```
"@@||fonts.gstatic.com",
```
that's excluding `fonts.gstatic.com` explictly, hence I think it's a good idea to prepend "User Rules for PAC" may be better than appending.

Hope there will be a fix for that, as I have to manually create my PAC file as a workaround. thanks in advance.
